### ISSUE TITLE
Render body as static markup

### ISSUE THREAD

**Issue by glennreyes** [iansinnott/react-static-webpack-plugin#8]
According to [ReactDOMServer.renderToStaticMarkup](https://facebook.github.io/react/docs/top-level-api.html#reactdomserver.rendertostaticmarkup):

> Similar to renderToString, except this doesn't create extra DOM attributes such as data-react-id, that React uses internally. This is useful if you want to use React as a simple static page generator, as stripping away the extra attributes can save lots of bytes.

Which in my opinion should be used instead of `renderToString` in the body as well.

In my opinion, `data-react-id` shouldn't be used externally anyway and removing it from the build, reduces the markup dramatically.

I would make a PR if we intend to do this! 🙌

**Comment by iansinnott** [comment]
Hey @glennreyes, I think it would be great to support this as an option. Something like:

```js
ReactStaticPlugin({
  routes: './client/routes.js',
  template: './template.js',
  renderToStaticMarkup: true, // <-- optionally do not use renderToString
})
```

The key though is that it would be _optional_. When I was evaluating the static site landscape I found that some projects did exactly that, used `renderToStaticMarkup`, and it didn't suit my needs. 

You are right that it increases the number of bytes that must be sent over the wire because of `data-react-id`, but it also limits the work react has to do on initial render when you call `ReactDOM.render(<App />, document.getElementById('root'));`. The reason being, React sees that the DOM has already been rendered and simply attaches event listeners.

From [the docs](https://facebook.github.io/react/docs/top-level-api.html#reactdomserver.rendertostring):

>If you call ReactDOM.render() on a node that already has this server-rendered markup, React will preserve it and only attach event handlers, allowing you to have a very performant first-load experience.

If you use `renderToStaticMarkup` instead of `renderToString` you do not get that benefit. 

There is definitely a use case for `renderToStaticMarkup`, and it might in fact be exactly what some users (like yourself) prefer, so I agree it would be great to have this option. I'd be happy to accept a PR.

**Comment by glennreyes** [comment]
@iansinnott I didn't think about that pre-render cycle before I was filing this issue which makes a lot of sense to use `renderToString`! Well, now that's what I'd rather want than static markup.

But to have `renderToStaticMarkup` as an option would also be reasonable in some use cases.

**Comment by iansinnott** [comment]
Cool, I'll close this out for now then. We can always reopen if someone comes along who prefers static markup.

**Comment by CedarDCX** [comment]
Having the option to render static markup would be something useful for us.

We're basically using this plugin to generate static pages - so static we don't even include the app.js bundle on the built files! Really, we're just using the plugin to allow us to use React as a templating engine and these static templates are then implemented into a CMS. It would be great if we could generate static markup without the react-* attributes, given they're redundant in this instance.

**Comment by iansinnott** [comment]
Thanks for the feeback @CedarDCX. That use case makes sense, so I think something like a `renderToStaticMarkup` option would be great. 

I will get around to implementing this eventually but I would be happy to take a PR 😃 . Here's the code in question: https://github.com/iansinnott/react-static-webpack-plugin/blob/master/src/index.js#L261

**Comment by CedarDCX** [comment]
PR submitted!

**Comment by iansinnott** [comment]
Merged. Thanks @CedarDCX. Closing this one out. Will be released as 1.6.
